lib/services/local/notifications.go

```diff
@@ -0,0 +1,307 @@
+	globalNotificationService       *generic.ServiceWrapper[*notificationsv1.GlobalNotification]
+	userNotificationStateService    *generic.ServiceWrapper[*notificationsv1.UserNotificationState]
+	userLastSeenNotificationService *generic.ServiceWrapper[*notificationsv1.UserLastSeenNotification]
+}
+
+// NewNotificationsService returns a new instance of the NotificationService.
+func NewNotificationsService(backend backend.Backend) (*NotificationsService, error) {
+	userNotificationService, err := generic.NewServiceWrapper[*notificationsv1.Notification](backend, types.KindNotification, notificationsUserSpecificPrefix, services.MarshalNotification, services.UnmarshalNotification)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	globalNotificationService, err := generic.NewServiceWrapper[*notificationsv1.GlobalNotification](backend, types.KindGlobalNotification, notificationsGlobalPrefix, services.MarshalGlobalNotification, services.UnmarshalGlobalNotification)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	userNotificationStateService, err := generic.NewServiceWrapper[*notificationsv1.UserNotificationState](backend, types.KindUserNotificationState, notificationsStatePrefix, services.MarshalUserNotificationState, services.UnmarshalUserNotificationState)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	userLastSeenNotificationService, err := generic.NewServiceWrapper[*notificationsv1.UserLastSeenNotification](backend, types.KindUserLastSeenNotification, notificationsUserLastSeenPrefix, services.MarshalUserLastSeenNotification, services.UnmarshalUserLastSeenNotification)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &NotificationsService{
+		userNotificationService:         userNotificationService,
+		globalNotificationService:       globalNotificationService,
+		userNotificationStateService:    userNotificationStateService,
+		userLastSeenNotificationService: userLastSeenNotificationService,
+	}, nil
+}
+
+// ListNotificationsForUser returns a paginated list of notifications which match a user, including both user-specific and global ones.
+func (s *NotificationsService) ListNotificationsForUser(ctx context.Context) ([]*notificationsv1.Notification, string, error) {
+	// TODO: rudream - implement listing notifications for a user with filtering/matching
+	return []*notificationsv1.Notification{}, "", nil
+}
+
+// CreateUserNotification creates a user-specific notification.
+func (s *NotificationsService) CreateUserNotification(ctx context.Context, username string, notification *notificationsv1.Notification) (*notificationsv1.Notification, error) {
+	if err := services.ValidateNotification(notification); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	notification.Kind = types.KindNotification
+	notification.Version = types.V1
+
+	if notification.Spec == nil {
+		notification.Spec = &notificationsv1.NotificationSpec{}
+	}
```

Comment:
It looks like `services.ValidateNotification` already rejects notifications with no `Spec`. That validation makes this conditional unnecessary, doesn't it?